### PR TITLE
[WIP] Adds wallet properties implementation

### PR DESCRIPTION
**Pull request by NejcZdovc**
## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: browser/payments/payments_service_impl.cc [hunk 1]
```diff
@@ -238,6 +238,10 @@ void PaymentsServiceImpl::OnWalletCreated(ledger::Result result) {
   TriggerOnWalletCreated(result);
 }
 
+void PaymentsServiceImpl::OnWalletProperties(ledger::WalletProperties result) {
```
Comment (bridiver) [inline browser/payments/payments_service_impl.cc]: we don't want to expose `ledger` outside of the payments service. What exactly is in wallet properties?

Comment (NejcZdovc) [reply browser/payments/payments_service_impl.cc]: this are the fields https://github.com/brave-intl/bat-native-ledger/pull/26/files#diff-2779bd349aa484b3cafcf4c1a138efd5R22. We use them all in UI, like balance, grants, range and choices for monthly contribution

Comment (bridiver) [reply browser/payments/payments_service_impl.cc]: I think we have individual methods for the balance and we should add them for anything else you need

Comment (bridiver) [reply browser/payments/payments_service_impl.cc]: but we should never directly expose anything in `ledger`. For Grants you might want to do something like `ContentSite` to expose whatever fields we need in the UI

Comment (bridiver) [reply browser/payments/payments_service_impl.cc]: basically we should only expose the things in payment service that we actually use in the brave rewards ui and leave all the internal details inside of ledger

Comment (bridiver) [reply browser/payments/payments_service_impl.cc]: if any kind of translation needs to be made to put it in the format we want for the user, that translation should happen in the payment service

File: browser/ui/webui/brave_rewards_ui.cc [hunk 2]
```diff
@@ -91,6 +109,29 @@ void RewardsDOMHandler::OnWalletCreateFailed() {
...
+      auto* render_view_host = web_contents->GetRenderViewHost();
+      if (render_view_host) {
+        render_view_host->SetWebUIProperty("rewards.walletInfo.probi", result.probi_);
+        render_view_host->SetWebUIProperty("rewards.walletInfo.balance", std::to_string(result.balance_));
+        for (auto const& rate : result.rates_) {
+          render_view_host->SetWebUIProperty("rewards.walletInfo.rates." + rate.first , std::to_string(rate.second));
+        }
+      }
+    }
+    web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.walletProperties");
```
Comment (bridiver) [inline browser/ui/webui/brave_rewards_ui.cc]: wouldn't this be a generic method to update the state with the new webui props?

File: components/brave_rewards_ui/reducers/rewards_reducer.ts [hunk 3]
```diff
@@ -28,6 +28,15 @@ const rewardsReducer: Reducer<Rewards.State | undefined> = (state: Rewards.State
       state = { ...state }
       state.walletCreateFailed = true
       break
+    case types.GET_WALLET_PROPERTIES:
+      chrome.send('getWalletProperties', [])
+      break
+    case types.ON_WALLET_PROPERTIES:
```
Comment (bridiver) [inline components/brave_rewards_ui/reducers/rewards_reducer.ts]: I see, so you're treating the response as an action to update the store? That seems like a reasonable route to go as well. I was thinking thinking something more like writing webui properties as `store.propertyname` and using a generic mechanism to write those to the actual store for the page, but either way works.

Comment (NejcZdovc) [reply components/brave_rewards_ui/reducers/rewards_reducer.ts]: so to basically have 1:1 mapping with redux store? this is how I did so far https://github.com/brave/brave-core/pull/293/files#diff-aa3505dbf36b5d03d8ba0751e0c99904R124 which I think it's quite the same as you were thinking.

File: browser/ui/webui/brave_rewards_ui.cc (hunk 2 above)
Comment (bridiver) [reply browser/ui/webui/brave_rewards_ui.cc]: we can also keep the balance update timers in c++ and just update the props

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: we talked about this with @bbondy and we decided to have all timers in app and only do calls to library when needed

Comment (bridiver) [reply browser/ui/webui/brave_rewards_ui.cc]: well, I'm not even a big fan of the fact that the balance update runs continuously when the payments ui is active and we definitely don't want it running all the time. I think it probably makes  sense to put that inside the webcontents observer though and not JS

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: balance is running only when you are on rewards page (chrome://rewards). We are already quite smart about it, so for example when you claim grant we call balance etc, so maybe we can have this conversation again and increase timer from a minute to something higher

**Comment by NejcZdovc** [comment]
closing in favour of #241

**Pull request closed without merging**